### DISCUSSION TITLE
What's the level of support for Vue 2?

### DISCUSSION THREAD

**Discussion by rchl** [vuejs/language-tools#740] [Q&A] [answered]
I've tried this with a Vue 2 (Nuxt 2) project using the Options API. I have [followed the readme](https://github.com/johnsoncodehk/volar/tree/master/extensions/vscode-vue-language-features#using) to enable experimental Vue 2 support and noticed that it doesn't seem to be picking types that are declared in the `types` array in `tsconfig.json`. 

For example the types for the `$t` function from `vue-i18n` are not taken into account in the template here:

![Screenshot 2021-11-27 at 15 30 11](https://user-images.githubusercontent.com/153197/143685641-796894b3-b0f4-4a52-9471-dc1d626d7a5b.png)

The same affects the script tag.

Is this expected with the current level of support or should I report it as an issue?

**Comment by johnsoncodehk** [comment, marked as answer]
Sorry for late to reply! vue-i18n already provided global properties types for vue2, to include it, you need to config `"types": [..., "vue-i18n/types"]` to `compilerOptions` in tsconfig, and export the component by `Vue.extend` or `defineComponent` from `@vue/composition-api`.

```json
{
  "compilerOptions": {
    // ...
    "types": [..., "vue-i18n/types"]
  },
  // ...
}
```

```html
<template>{{ $t }}</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({ ... });
</script>
```

or

```html
<template>{{ $t }}</template>

<script lang="ts">
import { defineComponent } from '@vue/composition-api';

export default defineComponent({ ... });
</script>
```

**Reply by rchl** [reply]
I was testing this in a Nuxt project with `nuxt-i18n` module that was registered in `types` array. So that's equivalent to registering `vue-i18n` itself.

I believe my issue had to do more with the vue2-specific compat that I've tried to clarify later in https://github.com/johnsoncodehk/volar/pull/750 (have to get back to it later, sorry)
